pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java

```diff
@@ -320,6 +322,33 @@ public String getPersistenceNamingEncoding() {
         }
     }
 
+    public static TopicName fromPersistenceNamingEncoding(String name) throws Exception {
+        if (name == null) {
+            return null;
+        }
+        final String[] arr = name.split("/");
+
+        if (arr.length == 4) {
+            String tenant = arr[0];
+            String namespacePortion = arr[1];
+            String domain = arr[2];
+            String encodedLocalName = arr[3];
+            final String decodedName = Codec.decode(encodedLocalName);
```

Comment:
Looks this case does not cover by any test? Could you please also add a test for this case?